include/deal.II/base/parameter_handler.h

```diff
@@ -1043,7 +1048,27 @@ class ParameterHandler : public Subscriptor
    * input.
    */
   virtual void
-  parse_input_from_json(std::istream &input, const bool skip_undefined = false);
+  parse_input_from_json(std::istream &input,
+                        const bool    skip_undefined               = false,
+                        const bool    assert_all_entries_are_found = false);
+
+  /**
+   * Parse input from a specified @param parameter_file independently of the type
+   * of input file (prm, xml, json) being used. The user can specify whether
```

Comment:
Maybe one thing you could add here is to say that this new function explicitly goes by the filename extension - in the specific functions, I could run `parse_input_from_json` with a file called `file.prm` and everything worked if the *content* was json - such a mistake is certainly not good style, but it happens especially for things that get passed on between different people with different habits. For those folks, the comment would make sure that the selection is by the filename extension and not by the format in the content (which could technically also be done but we do not want to implement it).

Reply:
I added this. Regarding the documentation, I added documentation to those functions that also explained the parameter `skip_undefined` that has been there before, while I did not add documentation to those function that also do not explain `skip_undefined`.